javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/TypeExtractor.java

```diff
@@ -638,10 +644,14 @@ public ResolvedType visit(FieldDeclaration node, Boolean solveLambdas) {
     }
     
     private static int getParamPos(Node node) {
-        if (demandParentNode(node) instanceof MethodCallExpr) {
-            MethodCallExpr call = (MethodCallExpr) demandParentNode(node);
+        if (demandParentNode(node, IS_NOT_ENCLOSED_EXPR) instanceof MethodCallExpr) {
+            MethodCallExpr call = (MethodCallExpr) demandParentNode(node, IS_NOT_ENCLOSED_EXPR);
             for (int i = 0; i < call.getArguments().size(); i++) {
-                if (call.getArguments().get(i) == node) return i;
+                Expression expression = call.getArguments().get(i);
+                while (expression instanceof EnclosedExpr) {
+                    expression = ((EnclosedExpr) expression).getInner();
```

Comment:
Here argument "expression" is an expression so you can write n.isEnclosedExpr()

Reply:
Good idea, agreed.